middlewares/logger_test.go

```diff
@@ -47,7 +48,9 @@ func TestLogger(t *testing.T) {
 		logfilePath = filepath.Join("/tmp", logfileName)
 	}
 
-	logger = NewLogger(logfilePath)
+	fi, _ := os.OpenFile(logfilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
```

Comment:
Don't we need to `defer` a file removal right after the file has been opened (without errors)?